api.js

```diff
@@ -27,10 +27,18 @@ const util = require('./util');
 const { isFunction } = require('./util/types');
 
 class Api extends EventEmitter {
-  constructor (provider, allowSubscriptions = true, middlewareClass) {
+  constructor (_provider, allowSubscriptions = true, middlewareClass) {
     super();
 
-    if (!provider || !isFunction(provider.send)) {
+    let provider = _provider;
```

Comment:
Seems that you don't actaully need to rename the param and that line is redundant.